client/helpers.go

```diff
@@ -79,38 +80,34 @@ func changeTargetsDelegation(repo *tuf.Repo, c changelist.Change) error {
 		if err != nil {
 			return err
 		}
-		r, _, err := repo.GetDelegation(c.Scope())
-		if _, ok := err.(data.ErrNoSuchRole); err != nil && !ok {
-			// error that wasn't ErrNoSuchRole
+		_, err = repo.GetDelegationRole(c.Scope())
+		if _, ok := err.(data.ErrInvalidRole); err != nil && !ok {
```

Comment:
I'm thinking we should have a different error to indicate the path to the parent was correct but the role didn't exist when we got there. `ErrInvalidRole` could indicate a bunch of problems other than non-existence.